### ISSUE TITLE
Analytics Manager:  Implement centralized event-handling on the client

### ISSUE THREAD

**Issue by kevinp** [DrivingThru/DrivingThruiOS#92]
@vfxdrummer  wrote: "We should build an analytics manager to register our events, so that we are able to easily send them to any analytics Service from one central spot."

Ideally, the analytics manager subscribes to events happening through the UI.  The event handlers are all in the analytics manager.  Let's keep the interface of the messages thin... so just identifiers and dates maybe?  And then if we need additional information when we push out to external services, we'll have an explicit data access call.

Longer term, I'm thinking a pub sub implementation
https://cloud.google.com/pubsub/

**Comment by kevinp** [comment]
@vfxdrummer , could you provide an estimate of effort (hours) required in the title?  I'm trying to get a rough idea of remaining effort and priorities.  Thanks!

**Comment by vfxdrummer** [comment]
I built an AnalyticsManager to the project.

It has a log method that takes DTAnalyticsEvent objects
DTAnalyticsEvent is an enum, where we can add events for everything that we track w/ optional arg.
I added these dummy ones as an example :
    ```case applicationOpened
    case example1
    case example2(argument1: String, argument2: String)```
So, we just need a list of what we’d like to initially track and it should be easy to add the events.
Right now, our log command calls through to Analytics.logEvent, the Firebase / GoogleAnalytics method that adds the event to the Firebase Analytics Dashboard.
